lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.ts

```diff
@@ -16,66 +16,85 @@
-    private apiService: AlfrescoApiService,
-    private dialog: MatDialog) {
-  }
+    sourceNodeNotFound = false;
 
-  openUploadFileDialog(currentFolderId?: string, selectionMode?: string, isAllFileSources?: boolean, restrictRootToCurrentFolderId?: boolean): Observable<Node[]> {
-    const select = new Subject<Node[]>();
-    select.subscribe({
-      complete: this.close.bind(this)
-    });
-    const data = <ContentNodeSelectorComponentData> {
-      title: 'Select a file',
-      actionName: NodeAction.ATTACH,
-      currentFolderId,
-      restrictRootToCurrentFolderId,
-      select,
-      selectionMode,
-      isSelectionValid: (entry: Node) => entry.isFile,
-      showFilesInResult: true,
-      showDropdownSiteList: false,
-      showLocalUploadButton: isAllFileSources
-  };
-    this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
-    return select;
-  }
+    constructor(
+        private apiService: AlfrescoApiService,
+        private logService: LogService,
+        private notificationService: NotificationService,
+        private dialog: MatDialog) {
+    }
+
+    openUploadFileDialog(currentFolderId?: string, selectionMode?: string, isAllFileSources?: boolean, restrictRootToCurrentFolderId?: boolean): Observable<Node[]> {
+        const select = new Subject<Node[]>();
+        select.subscribe({
+        complete: this.close.bind(this)
+        });
+        const data = <ContentNodeSelectorComponentData> {
+        title: 'Select a file',
+        actionName: NodeAction.ATTACH,
+        currentFolderId,
+        restrictRootToCurrentFolderId,
+        select,
+        selectionMode,
+        isSelectionValid: (entry: Node) => entry.isFile,
+        showFilesInResult: true,
+        showDropdownSiteList: false,
+        showLocalUploadButton: isAllFileSources
+    };
+        this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
+        return select;
+    }
 
-    async fetchNodeIdFromRelativePath(alias: string, opts: { relativePath: string }): Promise<string> {
-        const relativePathNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias, opts)
-        .catch((err) => this.handleError(err));
-        return relativePathNodeEntry?.entry?.id;
+    fetchNodeFromRelativePath(alias: string, opts: { relativePath: string }): Promise<NodeEntry> {
+        return from(this.apiService.nodesApi.getNode(alias, opts)).pipe(
+            catchError((error) => {
+                this.sourceNodeNotFound = true;
+                this.handleError(error);
+                return of(<NodeEntry> {});
+            })
+        ).toPromise();
     }
 
-    async fetchAliasNodeId(alias: string): Promise<string> {
-        const aliasNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias)
-        .catch((err) => this.handleError(err));
-        return aliasNodeEntry?.entry?.id;
+    fetchAliasNode(alias: string): Promise<NodeEntry> {
```

Comment:
Why do we need to change this method to return the Node instead of the nodeId that it was returning before? What is the benefit of it, are we solving any problem by this?

Reply:
rollbacked